Slow down the worker thread when API quota exceded

apisarenco opened the pull request:
Currently, individual jobs are postponed in execution, but workers have a full queue that they're constantly trying to send, thus negating the brake system put in place to slow them down when this happens.
This ensures the thread(s) are also waiting, to eliminate the chance that the downloader will ultimately fail. As long as there are not many parallel threads working.

Commit:
Slow down the worker thread when API quota exceded

Currently, individual jobs are postponed in execution, but workers have a full queue that they're constantly trying to send, thus negating the brake system put in place to slow them down when this happens.
This ensures the thread(s) are also waiting, to eliminate the chance that the downloader will ultimately fail. As long as there are not many parallel threads working.

martin-loetzsch commented:
I'm on vacation until September. If you need a review quickly, please ask
Ralph Kiers from Catawiki

On Thu, Aug 23, 2018, 12:14 PM Alex <notifications@github.com> wrote:

> @apisarenco <https://github.com/apisarenco> requested your review on:
> mara/facebook-ads-performance-downloader#23
> <https://github.com/mara/facebook-ads-performance-downloader/pull/23>
> Slow down the worker thread when API quota exceded.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/mara/facebook-ads-performance-downloader/pull/23#event-1804877001>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABULelBl0EQYioAIkWN04R3e9qCfqQcRks5uToCPgaJpZM4WJPRV>
> .
>

Commit:
Only block workers on rate limiting error

When Facebook gets too many requests it throws a rate limiting exception. Block the workers when this (and only this) exception is raised.

apisarenco commented:
I finally got to update this. This is also rebased on master. The slow-down now works only when the exception is Rate Limiting. Please someone review this.

martin-loetzsch reviewed (approved):
LGTM
